chore: omit 'docfx' from docs-presubmit build

tseaver opened the pull request:
We don't have a 'docfx' session in our noxfile (see #822).

Toward #797.

tseaver committed:
chore: omit 'docfx' from docs-presubmit build

We don't have a 'docfx' session in our noxfile (see #822).

Toward #797.

tseaver commented:
Note that I don't actually expect this PR to fix the `docs-presubmit` build, which has been failing in the `.kokoro/bulid.sh` setup before ever getting to the point of running `nox`.

dandhlee commented on .kokoro/build.sh:
```diff
@@ -31,7 +31,7 @@ env | grep KOKORO
 
 # Setup project id.
-export PROJECT_ID=$(cat "${KOKORO_GFILE_DIR}/project-id.txt")
+export PROJECT_ID=$(cat "${KOKORO_GFILE_DIR}/project-id.json")
```
`cat: /tmpfs/src/gfile/project-id.json: No such file or directory` I think it was fine before when you used `project-id.txt`?

tseaver replied:
@dandhlee That filename works for the main Kokoro build, but *not* for the `docs-presubmit` build (see [this failure](https://source.cloud.google.com/results/invocations/a3937c80-4f2f-48a0-a9cf-d6dfa090d34a/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fgoogle-auth-library-python%2Fdocs%2Fdocs-presubmit/log)).

Maybe there is something weird in the `google3` Kokoro config for this repository?  I honestly can't figure out why `docs-presubmit` is failing here.

tseaver committed:
chore: add additional 'gfile_resources' key

Our '.kokoro/build.sh' expects to find the files there, even for the
docs build.

tseaver committed:
chore: get 'gcloud' SDK installed for 'docs-presubmit'

Fingers crossed that I have the incantation correct -- I will fix the
owlbot replacment once it works.

tseaver committed:
chore: sigh

tseaver committed:
chore: heavy sigh

tseaver committed:
chore: asthmatic wheeze

dandhlee commented on owlbot.py:
```diff
@@ -7,11 +7,41 @@
+  && sudo apt-get install google-cloud-sdk
+
+CMD ["python3.8"]""",
+    )
```
Adding the last bit to change `@nox.session(python="3.7")` to `@nox.session(python="3.8")` for `docs(session)` on noxfile.py should get docs session to run successfully now!

tseaver replied:
I fixed that in 0c44e6a by modifying `noxfile.py` directly, because `owlbot.py` skips over it.

tseaver committed:
chore: Cheyne-Stokes respiration

tseaver committed:
chore: death rattle

dandhlee commented on .kokoro/docker/docs/Dockerfile:
```diff
@@ -64,4 +64,12 @@ RUN wget -O /tmp/get-pip.py 'https://bootstrap.pypa.io/get-pip.py' \
+  && curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | \
+    apt-key --keyring /usr/share/keyrings/cloud.google.gpg  add - \
+  && apt-get update -y \
+  && apt-get install python2 google-cloud-sdk -y
```
do we need `python2` if we don't need to run system tests?

tseaver replied:
This PR isn't trying to fix #832 (at least not yet).  To do that, we need either to have a `.kokoro/build-systests.sh` script, which runs only the systests, or else work out some environment variable which we use to turn them on / off in `.kokoro/build.sh`.

Either way will require adding something like `.kokoro/presubmit/system-3.8.cfg`, to either set the environment variable or trigger the alternate script, which will then need a corresponding tweak to the `google3` Kokoro configuration for this repostitory.

dandhlee replied:
Currently it just skips the system tests (missing 3.7 interpreter), and `docs` session runs fine. I think it's fine to move forward as-is.

tseaver replied:
PR #833 is based on this PR:  once we merge here, I will rebase it against `master`, and then wait to merge until the `google3` changes are in place to enable the `Kokoro system-3.7` build.

dandhlee commented:
Think we're almost there. Blocked by system tests but it'll be fixed soon, docs-presubmit is happy without `docfx` and runs `docs` session fine. Not sure why OwlBot isn't happy though.

tseaver commented:
@dandhlee Owlbot gets its nose in a twist when the PR is rebased to remove its earlier, unwanted commits.

dandhlee commented:
Hm, I thought rerunning  it would resolve itself but I guess not.

tseaver commented:
@dandhlee I think we need to get a refresh-token PR merged to `master` and then here, first.

tseaver commented:
Blocked on #834.

busunkim96 committed:
chore: update secrets (#834)

tseaver commented:
@dandhlee PTAL.

tseaver commented:
Added `do not merge` label pending the `google3` change to activate the `Kokoro system-3.7` build.

tseaver commented:
Oops, the Owlbot failure is real:  the substitution I added breaks when I run synthool locally.

dandhlee reviewed (approved):
LGTM.

tseaver committed:
chore: agonal respiration

tseaver committed:
Merge branch 'master' into 797-drop-docfx-from-docs-presubmit-build
